cmd/autoscaler/main.go

```diff
@@ -98,18 +107,6 @@ func main() {
 
 	ctx, informers := injection.Default.SetupInformers(ctx, cfg)
 
-	kubeClient := kubeclient.Get(ctx)
-
-	// We sometimes startup faster than we can reach kube-api. Poll on failure to prevent us terminating
-	if perr := wait.PollImmediate(time.Second, 60*time.Second, func() (bool, error) {
-		if err = version.CheckMinimumVersion(kubeClient.Discovery()); err != nil {
-			log.Printf("Failed to get k8s version %v", err)
-		}
-		return err == nil, nil
-	}); perr != nil {
-		log.Fatal("Timed out attempting to get k8s version: ", err)
-	}
```

Comment:
If you'd left this as is, then you could save some `kubeclient.Get` calls below.